### ISSUE TITLE
Cache key behaviour for alliased timestamps

### ISSUE THREAD

**Issue by alexandru-calinoiu** [rails/rails#26417]
### Steps to reproduce

I a simple model

``` ruby
class Epixcms::Movies::Movie < ActiveRecord::Base
  alias_attribute :updated_at, :modified_date
  alias_attribute :created_at, :create_date
end

movie = Epixcms::Movies::Movie.new
movie.modified_date = Time.zone.now
movie.cache_key
```
### Expected behavior

It should respect the defined alias and include the modified_date into the cache_key
### Actual behavior

It does not include the modified_date
### System configuration

Rails 4.2.7.1
Ruby 2.3.0

**Comment by prathamesh-sonpatki** [comment]
Shouldn't it be `alias_attribute :modified_date, :updated_at` ? new attribute name comes first.

**Comment by alexandru-calinoiu** [comment]
In this case the `:updated_at` is the new attribute, as it does not exist the legacy database has `modified_date` column.

**Comment by alexandru-calinoiu** [comment]
There is also a subtle difference in behavior for this method that is annoying.

Supposedly I work around my issue by writing this:

```
movie.cache_key(:modified_date)
```

When modified_date is nil it will throw an exception. But

```
movie.cache_key
```

When updated_at is nil does not fail it returns a cache key with the id only.

I will expect to have have the same behavior for nil modified_date.

**Comment by prathamesh-sonpatki** [comment]
Can you please post what exception you got?

**Comment by alexandru-calinoiu** [comment]
It will be a nil exception on this line https://github.com/rails/rails/blob/b326e82dc012d81e9698cb1f402502af1788c1e9/activerecord/lib/active_record/integration.rb#L61

**Comment by matthewd** [comment]
Given the surrounding conditionals, the nil exception definitely sounds like a bug.

Can you reproduce the alias-ignoring behaviour on 5.0 or master? I think it might be fixed there.

**Comment by prathamesh-sonpatki** [comment]
Opened https://github.com/rails/rails/pull/26425 to fix the issue of getting an error when named timestamp column has value `nil`.

**Comment by prathamesh-sonpatki** [comment]
Found out that the issue related to aliased attribute exists on master as well. Reproduction script here - https://gist.github.com/prathamesh-sonpatki/1c82c0a86d1ca171b086324d41af8491

I fixed it here - https://github.com/prathamesh-sonpatki/rails/commit/78c8235a5768b114532d64d41db0307d6a5a7f9e

but would like to get some feedback cc @sgrif @matthewd 

Also should we fix this issue for `write_attribute` method as well? It also does not honor aliased attributes as of now.
